mido/midifiles/units.py

```diff
@@ -12,35 +12,31 @@ def tick2second(tick, ticks_per_beat, tempo):
 def second2tick(second, ticks_per_beat, tempo):
     """Convert absolute time in seconds to ticks.
 
-    Returns absolute time in ticks for a chosen MIDI file time
-    resolution (ticks per beat, also called PPQN or pulses per quarter
-    note) and tempo (microseconds per beat).
+    Returns absolute time in ticks for a chosen MIDI file time resolution
+    (ticks/pulses per quarter note, also called PPQN) and tempo (microseconds
+    per quarter note). Normal rounding applies.
     """
     scale = tempo * 1e-6 / ticks_per_beat
-    return second / scale
+    return int(round(second / scale))
 
 
-def bpm2tempo(bpm):
-    """Convert beats per minute to MIDI file tempo.
+def bpm2tempo(bpm, time_signature=(4, 4)):
+    """Convert BPM (beats per minute) to MIDI file tempo (microseconds per
+    quarter note).
 
-    Returns microseconds per beat as an integer::
-
-        240 => 250000
-        120 => 500000
-        60 => 1000000
+    Depending on the chosen time signature a bar contains a different number of
+    beats. These beats are multiples/fractions of a quarter note, thus the
+    returned BPM depend on the time signature. Normal rounding applies.
     """
-    # One minute is 60 million microseconds.
-    return int(round((60 * 1000000) / bpm))
-
+    return int(round(60 * 1e6 / bpm * time_signature[1] / 4.))
 
-def tempo2bpm(tempo):
-    """Convert MIDI file tempo to BPM.
 
-    Returns BPM as an integer or float::
+def tempo2bpm(tempo, time_signature=(4, 4)):
+    """Convert MIDI file tempo (microseconds per quarter note) to BPM (beats
+    per minute).
 
-        250000 => 240
-        500000 => 120
-        1000000 => 60
+    Depending on the chosen time signature a bar contains a different number of
+    beats. These beats are multiples/fractions of a quarter note, thus the
+    returned tempo depends on the time signature.
```

Comment:
I think it is good to mention here that what constitutes a beat is defined by the denominator of the time signature, and give an example: in 2/4, 3/4, 4/4, ... the beat is a quarter note. In 3/8, 6/8, ... the beat is an eight note, etc.